pkg/util/tracing/span.go

```diff
@@ -91,6 +91,17 @@ func (sp *Span) Finish() {
 	sp.i.Finish()
 }
 
+// FinishAndGetRecording finishes the span and gets a recording at the same
+// time. This is offered as a combined operation because, otherwise, the caller
+// would be forced to collect the recording before finishing and so the span
+// would appear to be unfinished in the recording (it's illegal to collect the
+// recording after the span finishes, except by using this method).
+func (sp *Span) FinishAndGetRecording(recType RecordingType) Recording {
+	sp.Finish()
+	// Reach directly into sp.i to avoide the done() check in sp.GetRecording().
```

Comment:
Did you mean to add a done check in sp.GetRecording below? (Or am I not seeing it from elsewhere?)

Reply:
not yet, cause I think there's other code paths that rely on collecting after Finish that I haven't fixed yet. But it's coming.